### PR TITLE
add javadoc and make some api changes to make more friendly to java

### PR DESCRIPTION
**Pull request by jhouserizer**
add javadoc and make some api changes to make more friendly to java idioms

### CODE REVIEW

**Review by GaryWKeim — APPROVED** [review]

File: tcstore-exporter/src/main/java/org/terracotta/store/export/ParquetOptions.java [hunk 1]
```diff
@@ -62,13 +70,34 @@
     private Integer maxStringLength = -1;
     private Integer maxByteArraySize = -1;
 
+    /**
+     * Add a cell to the "whitelist" of cells that should be include from output
+     *
+     * If there is no whitelist or blacklist then all cells will be candidates for output
```
Comment (cschanck) [inline tcstore-exporter/src/main/java/org/terracotta/store/export/ParquetOptions.java:76]: Comment here or at the top that any failure here does not fail the process.

File: tcstore-exporter/src/main/java/org/terracotta/store/export/ParquetOptions.java [hunk 2]
```diff
@@ -62,13 +70,34 @@
...
+     *
+     * @param name the name of the cell
+     * @param type the Type of the cell
+     * @see #addBlackListCell(String, Type)
+     */
     public void addWhiteListCell(String name, Type type) {
         try {
```
Comment (cschanck) [inline tcstore-exporter/src/main/java/org/terracotta/store/export/ParquetOptions.java:83]: Alternatively, use Objects.requireNonNull(name); etc to validate input.

Comment (cschanck) [inline tcstore-exporter/src/main/java/org/terracotta/store/export/ParquetOptions.java:83]: addWhiteListCellDefinition

File: tcstore-exporter/src/main/java/org/terracotta/store/export/StoreExportException.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+/**
+ * StoreExportException
+ */
+package org.terracotta.store.export;
+
+
+public class StoreExportException extends Exception {
```
Comment (cschanck) [inline tcstore-exporter/src/main/java/org/terracotta/store/export/StoreExportException.java:7]: James javadoc fail.

**Pull request merged**